test/sharness/t0180-pubsub.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+test_description="Test dht command"
+
+. lib/test-lib.sh
+
+# start iptb + wait for peering
+NUM_NODES=5
+test_expect_success 'init iptb' '
+  iptb init -n $NUM_NODES --bootstrap=none --port=0
+'
+
+startup_cluster $NUM_NODES --enable-pubsub-experiment
+
+test_expect_success 'peer ids' '
+  PEERID_0=$(iptb get id 0) &&
+  PEERID_2=$(iptb get id 2)
+'
+
+# ipfs pubsub sub
+test_expect_success 'pubsub' '
+	echo "testOK" > expected &&
+	mkfifo done &&
+
+	(
+		ipfsi 0 pubsub sub --enc=ndpayload testTopic | 
+			while read line; do
+				echo $line > actual &&
+				echo > done
+				exit
```

Comment:
Yeah that is a bit hackish I agree. For some reason this loop even needs two runs, I didn't get why and I was glad it worked.